client/src/utils/inputValidator.ts

```diff
@@ -0,0 +1,30 @@
+import type { InputValidation } from 'types';
+
+import { VALIDATOR, VALIDATION_ERRORS } from '../constants';
+
+const inputValidator = (inputValue: string, inputName: string, validation?: InputValidation) => {
+  const value = inputValue.trim();
+
+  let isValid = true;
+  let message = '';
+
+  if (validation) {
+    const validators = Object.entries(validation);
+
+    isValid = !validators
+      .map(validator => {
+        const valid = VALIDATOR[validator[0]](value, validator[1]);
```

Comment:
please define what is validator[0] and what is validator[1] using variables and types

Reply:
```
const validationRule = validator[0] as InputValidatorKey;
const validationRuleValue = validator[1];
```
How about these?

Reply:
good, but can be even better :) 
`.map(([name1, name2]) => {`